src/game_config_view.cpp

```diff
@@ -49,27 +49,36 @@ config_array_view game_config_view::child_range(config_key_type key) const
 	return res;
 }
 
-const config& game_config_view::find_child(config_key_type key, const std::string &name, const std::string &value) const
+optional_const_config game_config_view::find_child(config_key_type key, const std::string &name, const std::string &value) const
 {
 	for(const config& cfg : cfgs_) {
-		if(const config& res = cfg.find_child(key, name, value)) {
+		if(optional_const_config res = cfg.find_child(key, name, value)) {
 			return res;
 		}
 	}
 	LOG_CONFIG << "gcv : cannot find [" << key <<  "] with " << name  << "=" << value << ", count = " << cfgs_.size();
 	const config cfg;
-	return cfg.child("invalid");
+	return optional_const_config();
 }
 
-const config& game_config_view::child(config_key_type key) const
+const config& game_config_view::manditory_child(config_key_type key) const
 {
 	for(const config& cfg : cfgs_) {
 		if(const auto res = cfg.optional_child(key)) {
 			return res.value();
 		}
 	}
-	const config cfg;
-	return cfg.child("invalid");
+	throw "missing anditory child";
```

Comment:
Missing M. Also should probably be a custom exception type instead of a `const char*`. If you can't be bothered, at least make it `std::string`.